### PR TITLE
Bug fixes:

### PR DESCRIPTION
**Pull request by wmathurin**
- smart sync explorer should show new image for locally created and updated record / deleted image for locally created or updated then deleted record
- sync up locally created then deleted records should not go to server
- smartsyncexplorer was not doing correct update/delete for locally created records

### CODE REVIEW

File: libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m [hunk 1]
```diff
@@ -716,7 +716,13 @@ - (void)resumeSyncUpOneEntry:(SFSyncState*)sync
             [target updateOnServer:objectType objectId:objectId fields:fields completionBlock:completeBlockUpdate failBlock:failBlockUpdate];
             break;
         case SFSyncUpTargetActionDelete:
-            [target deleteOnServer:objectType objectId:objectId completionBlock:completeBlockDelete failBlock:failBlockDelete];
```
Comment (wmathurin) [inline libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m]: Don't go to server when deleting locally created records (already on Android: https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/1159/commits/e2f720708069ecc083718f2e716f7083ffbed745)

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/ContactListViewController.m [hunk 2]
```diff
@@ -301,12 +301,15 @@ - (UIView *)accessoryViewForContact:(ContactSObjectData *)contact {
 
     if ([self.dataMgr dataHasLocalChanges:contact]) {
         UIImage *localImage;
-        if ([self.dataMgr dataLocallyCreated:contact])
```
Comment (wmathurin) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer/Classes/ContactListViewController.m]: Show deleted image if deleted (regardless of whether it was created or updated also)
Show created image if created but not deleted (regardless of whether it was also updated)

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorerCommon/SObjectDataManager.m [hunk 3]
```diff
@@ -172,13 +172,13 @@ - (void)createLocalData:(SObjectData *)newData {
 - (void)updateLocalData:(SObjectData *)updatedData {
     [updatedData updateSoupForFieldName:kSyncManagerLocal fieldValue:@YES];
     [updatedData updateSoupForFieldName:kSyncManagerLocallyUpdated fieldValue:@YES];
-    [self.store upsertEntries:@[ updatedData.soupDict ] toSoup:[[updatedData class] dataSpec].soupName withExternalIdPath:kSObjectIdField error:nil];
```
Comment (wmathurin) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorerCommon/SObjectDataManager.m]: It was wrong to do a upsert withExternalIdPath:kSObjectIdField - when the record is locally created, that field will be empty, as a result a new record is inserted!

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorerCommon/SObjectDataManager.m [hunk 4]
```diff
@@ -172,13 +172,13 @@ - (void)createLocalData:(SObjectData *)newData {
...
+    [self.store upsertEntries:@[ updatedData.soupDict ] toSoup:[[updatedData class] dataSpec].soupName];
 }
 
 - (void)deleteLocalData:(SObjectData *)dataToDelete {
     [dataToDelete updateSoupForFieldName:kSyncManagerLocal fieldValue:@YES];
     [dataToDelete updateSoupForFieldName:kSyncManagerLocallyDeleted fieldValue:@YES];
-    [self.store upsertEntries:@[ dataToDelete.soupDict ] toSoup:[[dataToDelete class] dataSpec].soupName withExternalIdPath:kSObjectIdField error:nil];
```
Comment (wmathurin) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorerCommon/SObjectDataManager.m]: It was wrong to do a upsert withExternalIdPath:kSObjectIdField - when the record is locally created, that field will be empty, as a result a new record is inserted!

**Comment by huminzhi** [comment]
test me

**Pull request merged**